src/compiler/checker.ts

```diff
@@ -16743,6 +16746,65 @@ namespace ts {
             }
         }
 
+        function markDecoratorMedataDataTypeNodeAsReferenced(node: TypeNode): void {
+            const entityNameOrToken = getEntityNameForDecoratoryMetadata(node);
+            if (entityNameOrToken && isEntityName(entityNameOrToken)) {
+                markEntityNameOrEntityExpressionAsReference(entityNameOrToken);
+            }
+        }
+
+        type voidUndefinedNullOrNeverTypeNode = Token<SyntaxKind.VoidKeyword | SyntaxKind.UndefinedKeyword | SyntaxKind.NullKeyword | SyntaxKind.NeverKeyword>;
+
+        function getEntityNameForDecoratoryMetadata(node: TypeNode): EntityName | voidUndefinedNullOrNeverTypeNode {
+            if (node) {
+                switch (node.kind) {
+                    case SyntaxKind.IntersectionType:
+                    case SyntaxKind.UnionType:

```

Comment:
we allow this if the Individual type leads to same identifier. This is to make sure string | null will emit type as string.. E | null results in E and "foo" | "bar"  results in string as type